### PR TITLE
grpc: Add keepalive Support

### PR DESCRIPTION
**Pull request by Jozott00**
**Subsystem**
gRPC

**Problem Description**
To reduce latency because of required reconnections to the server, the client sometimes wants to keep a connection open.

**Solution**
This can be done by setting the keepalive configuration option on the gRPC ClientConfiguration.

```kt
GrpcClient("localhost", 50051) {
    keepAlive {
         time = 10.seconds
         timeout = 20.seconds
         withoutCalls = false
    }
}
```

It also removes some tests that are either duplicates or testing undefined behavior (throwing in onClose of serverInterceptor).

### CODE REVIEW

File: grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcClient.kt [hunk 1]
```diff
@@ -294,4 +294,65 @@ public class GrpcClientConfiguration internal constructor() {
...
+     * Client authors must coordinate with service owners for whether a particular client-side
+     * setting is acceptable.
+     *
+     * @property time Specifies the maximum amount of time the channel can remain idle before a
+     * keep-alive ping is sent to the server to check the connection state.
+     * The default value is `Duration.INFINITE`, which disables keep-alive pings when idle.
+     *
+     * @property timeout Sets the amount of time to wait for a keep-alive ping response.
+     * If the server does not respond within this timeout, the connection will be considered broken.
+     * The default value is 20 seconds.
```
Comment (Mr3zee) [inline grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcClient.kt:345]: Any particular reason for 20 seconds?

**Review by Mr3zee — APPROVED** [review]
Great job!

File: grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcClient.kt (hunk 1 above)
Comment (Jozott00) [reply grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcClient.kt:345]: 20 seconds is defined by the gRPC specification https://grpc.io/docs/guides/keepalive/#keepalive-configuration-specification
